apps/etterlatte-brev-api/src/main/kotlin/no/nav/etterlatte/brev/oversendelsebrev/OversendelseBrevService.kt

```diff
@@ -0,0 +1,280 @@
+import no.nav.etterlatte.brev.model.BrevInnhold
+import no.nav.etterlatte.brev.model.BrevProsessType
+import no.nav.etterlatte.brev.model.Mottaker
+import no.nav.etterlatte.brev.model.OpprettNyttBrev
+import no.nav.etterlatte.brev.model.Pdf
+import no.nav.etterlatte.brev.model.Slate
+import no.nav.etterlatte.brev.model.Status
+import no.nav.etterlatte.brev.toMottaker
+import no.nav.etterlatte.libs.common.behandling.Klage
+import no.nav.etterlatte.libs.common.behandling.KlageUtfallMedData
+import no.nav.etterlatte.libs.common.behandling.SakType
+import no.nav.etterlatte.libs.common.feilhaandtering.GenerellIkkeFunnetException
+import no.nav.etterlatte.libs.common.feilhaandtering.UgyldigForespoerselException
+import no.nav.etterlatte.libs.common.person.Folkeregisteridentifikator
+import no.nav.etterlatte.libs.common.person.Vergemaal
+import no.nav.etterlatte.libs.common.tidspunkt.Tidspunkt
+import no.nav.etterlatte.token.BrukerTokenInfo
+import org.slf4j.LoggerFactory
+import java.time.LocalDate
+import java.util.UUID
+
+interface OversendelseBrevService {
+    fun hentOversendelseBrev(behandlingId: UUID): Brev?
+
+    suspend fun opprettOversendelseBrev(
+        behandlingId: UUID,
+        sakId: Long,
+        brukerTokenInfo: BrukerTokenInfo,
+    ): Brev
+
+    fun distribuerOversendelseBrev(
+        brevId: Long,
+        sakId: Long,
+        brukerTokenInfo: BrukerTokenInfo,
+    ): Brev
+
+    fun ferdigstillOversendelseBrev(
+        brevId: Long,
+        sakId: Long,
+        brukerTokenInfo: BrukerTokenInfo,
+    ): Pdf
+}
+
+class OversendelseBrevServiceImpl(
+    private val brevRepository: BrevRepository,
+    private val journalfoerBrevService: JournalfoerBrevService,
+    private val distribusjonService: DistribusjonService,
+    private val pdfGenerator: PDFGenerator,
+    private val sakService: SakService,
+    private val adresseService: AdresseService,
+    private val brevdataFacade: BrevdataFacade,
+) : OversendelseBrevService {
+    private val logger = LoggerFactory.getLogger(this::class.java)
+
+    override fun hentOversendelseBrev(behandlingId: UUID): Brev? {
+        return brevRepository.hentBrevForBehandling(behandlingId, Brevtype.OVERSENDELSE_KLAGE).firstOrNull()
+    }
+
+    override suspend fun opprettOversendelseBrev(
+        behandlingId: UUID,
+        sakId: Long,
+        brukerTokenInfo: BrukerTokenInfo,
+    ): Brev {
+        val eksisterendeBrev =
+            brevRepository.hentBrevForBehandling(behandlingId, Brevtype.OVERSENDELSE_KLAGE)
+                .firstOrNull()
+        if (eksisterendeBrev != null) {
+            return eksisterendeBrev
+        }
+
+        val sak = sakService.hentSak(sakId, brukerTokenInfo)
+        val generellBrevData =
+            brevdataFacade.hentGenerellBrevData(
+                sakId = sakId,
+                // Setter behandlingId som null for å unngå å hente en behandling med klageId'en
+                behandlingId = null,
```

Comment:
Høres ut som en todo her

Reply:
Du tenker på å være mer fleksibel på hvordan henting av generellBrevData håndterer behandlingId? Eller at det kunne vært en egen wrapper for klage?

Reply:
Ja, virker litt tvilsomt at behandlingId og klageId går om hverandre her. Men det er kanskje det som gir mest mening 🤔

Reply:
Ja, det er nok noe litt off. Men jeg tror det nok vil bli mer av ting som har "behandling"id'er som ikke er behandling direkte, som tilbakekreving og evt. andre behandlinger (refusjon av advokat f.eks.) på sikt. Så det blir å gjøre den tette koblingen opp mot behandling litt løsere etter hvert i brev-api